### PR TITLE
feat(smolvla): add configurable BF16 and FP32 model precision

### PR DESCRIPTION
**Pull request by adhanaku**
## Summary

Adds configurable numerical precision for SmolVLA models.

- Adds a precision option supporting `bfloat16` and `float32`.
- Passes the selected precision from the SmolVLA configuration through the policy and model initialization layers.
- Converts model parameters to the requested dtype.
- Keeps numerically sensitive vision, connector, and normalization layers in FP32 when using BF16.
- Preserves BF16 as the default behavior for backward compatibility.
- Enables `compress_to_fp16=True` for OpenVINO exports to reduce exported model size.

## Motivation

This allows SmolVLA to run in full FP32 when BF16 is unsupported or when higher numerical precision is needed, while retaining the lower-memory BF16 configuration as the default.

### CODE REVIEW

File: library/src/physicalai/policies/smolvla/model.py [hunk 1]
```diff
@@ -800,6 +816,34 @@ def set_requires_grad(self) -> None:
...
+        """
+        if precision == "float32":
+            self.to(dtype=torch.float32)
+            return
+
+        target_dtype = _resolve_precision_dtype(precision)
+        self.to(dtype=target_dtype)
+
+        params_to_keep_float32 = [
+            "vision_model",
```
Comment (sovrasov) [inline library/src/physicalai/policies/smolvla/model.py:836]: What is the source of this list? Does it match current implemenetion?

File: library/src/physicalai/policies/smolvla/policy.py [hunk 2]
```diff
@@ -876,7 +886,7 @@ def extra_export_args(self) -> dict[str, ExportParameters]:
         )
         extra_args["openvino"] = OpenVINOExportParameters(
             outputs=output_names,
-            compress_to_fp16=False,
+            compress_to_fp16=True,
```
Comment (sovrasov) [inline library/src/physicalai/policies/smolvla/policy.py:889]: Does this lead to any regression?

File: library/src/physicalai/policies/smolvla/model.py (hunk 1 above)
Comment (adhanaku) [reply library/src/physicalai/policies/smolvla/model.py:836]: This list is adapted from the existing Pi0.5 mixed-precision implementation, with the PaliGemma module names mapped to their SmolVLM equivalents. I validated it against the current pretrained SmolVLA implementation using and every selector matches at least one parameter

File: library/src/physicalai/policies/smolvla/policy.py (hunk 2 above)
Comment (adhanaku) [reply library/src/physicalai/policies/smolvla/policy.py:889]: No. Observed SR=62 with libero10 task suite for 40 episodes, for both FP32 and FP16 model

File: library/src/physicalai/policies/smolvla/policy.py [hunk 3]
```diff
@@ -336,6 +341,9 @@ def _initialize_model(
                     msg = f"  - {k}"
                     logger.warning(msg)
 
+            # Apply dtype/precision
+            self.model._model.to_bfloat16_for_selected_params(self.config.dtype)
+
             # Apply requires_grad
             self.model._model.set_requires_grad()  # noqa: SLF001
```
[the remark this answers is not in the document]
Comment (adhanaku) [reply library/src/physicalai/policies/smolvla/policy.py:348]: Added the missing import to resolve this

File: library/src/physicalai/policies/smolvla/config.py [hunk 4]
```diff
@@ -78,6 +79,8 @@ class SmolVLAConfig(SnapFlowConfigMixin, Config):
     inherited ``snapflow_*`` attributes.
     """
 
+    dtype: Literal["bfloat16", "float32"] = "bfloat16"
```
[the remark this answers is not in the document]
Comment (adhanaku) [reply library/src/physicalai/policies/smolvla/config.py:83]: Added the missing import to resolve this

File: library/src/physicalai/policies/smolvla/model.py [hunk 5]
```diff
@@ -53,6 +53,16 @@ def _lazy_import_transformers() -> tuple:
 
 logger = logging.getLogger(__name__)
 
+def _resolve_precision_dtype(precision: Literal["bfloat16", "float32"]) -> torch.dtype:
```
[the remark this answers is not in the document]
Comment (adhanaku) [reply library/src/physicalai/policies/smolvla/model.py:65]: Added the missing import to resolve this

File: library/src/physicalai/policies/smolvla/policy.py [hunk 6]
```diff
@@ -111,6 +112,7 @@ def __init__(  # noqa: PLR0913
         self,
         # Pretrained model id
         pretrained_name_or_path: str | Path | None = None,
+        dtype: Literal["bfloat16", "float32"] = "bfloat16",
```
[the remark this answers is not in the document]
Comment (adhanaku) [reply library/src/physicalai/policies/smolvla/policy.py:116]: Added the missing import to resolve this

**Review by sovrasov — APPROVED** [review]
